OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -285,13 +285,17 @@ public void onException(Exception ex) {
                             if (key.equalsIgnoreCase(fieldView.getField().getName())) {
                                 List<RegisterResponseFieldError> error = messageBody.get(key);
                                 showErrorOnField(error, fieldView);
-                                fieldErrorShown = true;
+                                if (!fieldErrorShown) {
```

Comment:
This flag's sole purpose is to avoid multiple dialogs from appearing and since its a small scoped local flag, its better to have a verbose name for better readability of code.